src/lib.rs

```diff
@@ -875,6 +875,29 @@ where
   }
 }
 
+/// Compute the circuit digest of a [StepCircuit].
+pub fn circuit_digest<
+  G1: Group<Base = <G2 as Group>::Scalar>,
+  G2: Group<Base = <G1 as Group>::Scalar>,
+  C: StepCircuit<G1::Scalar>,
+>(
+  circuit: &C,
+  is_primary_circuit: bool,
+) -> G1::Scalar {
+  let augmented_circuit_params =
+    NovaAugmentedCircuitParams::new(BN_LIMB_WIDTH, BN_N_LIMBS, is_primary_circuit);
```

Comment:
> the `is_primary_circuit` flag makes a ton of sense when you define a generic struct parameterized by a group `G` that can be used symmetrically to speak of `G1` or `G2`. In this case, the flag allows you to detect when that generic structure is being called with the primary group of the curve, and introduce an asymmetry of behavior within the generic function, for that case.
> 
> There’s probably a simpler way to proceed for the present function, isn’t there?

I don't see how; here we need the flag to tell us how to construct the circuit params

Reply:
Well, it seems that when we're considering a group pair, `G1` is by convention the primary. And the present function can't be called on anything but G1. 

But this is a nit, if your point is the function could theoretically be called with typed reversed, fine. Note we already hardcode this in the constructor of `PublicParams` though.